README.md

```diff
@@ -1,18 +1,116 @@
+[![Build Status](https://travis-ci.org/UCLA-CS130/Caviar-Hacker-Grid.svg?branch=master)](https://travis-ci.org/UCLA-CS130/Caviar-Hacker-Grid)
+
 # Caviar-Hacker-Grid
 CS130 Winter 2017 Webserver Project
 
 Commands are to be run from the base Caviar-Hacker-Grid directory.
+# Installation
+You must clone the repository recursively.
+``` 
+git clone --recursive https://github.com/UCLA-CS130/Caviar-Hacker-Grid.git 
+```
 
-To build program binaries & webserver:
-  make
+To run the server, cd into the directory and run 
+```
+make && ./webserver test_config
+```
 
-To build test:
+# Commands
+To run tests:
+  ```
   make test
+  ```
   
+To run integration test:
+  ```
+  make integration
+  ```
+
 To clean all binaries:
+  ```
   make clean
+  ```
  
-To run webserver
-  ./webserver <config_file>
   
-The webserver currently responds to echo requests and file requests.
+# Config File Format
+Conforms to the standard API:
+```
+port <port number>;
+
+path /path HandlerName {
+  HandlerOptions;
+}
+
+default defaultHandler {}
+```
+
+port and default are mandatory tags. There can be as many path handlers as you want.
+
+port must have a valid unused port that can be bound, otherwise the server will not begin. 
+
+Path handlers are longest prefixed matched, so you can have subpaths. Be careful to ensure that the paths have a slash prepended: otherwise, they won't ever be matched.
+
+The HandlerName is the name of the handler you want to use. If the handler is not found, the block is ignored.
+
+
+Currently, the supported handlers are:
+* NotFoundHandler
+  * No options
+* EchoHandler
+  * No options
+* StaticHandler
+  * Mandatory: 
+      * root <path_to_files_to_serve>
+        * Note that the path is relative to the location served from.
+        * This means you can take advantage of relative directories. 
+
+
+# Layout
+## Existing Code
+* `main.cc` : Gets and parses command lines arguments, passed to webserver class
+* `webserver.cc` : Parses and initializes the config for the server, including validation. This initializes the request handlers.
```

Comment:
We should consider merging the server and webserver files as we talked about. Unless you see a reason to keep them separate?